### PR TITLE
Animation state refactoring

### PR DESCRIPTION
**Pull request by gleblebedev**

### CODE REVIEW

File: Source/Urho3D/Graphics/AnimationController.h [hunk 1]
```diff
@@ -111,6 +123,12 @@ struct URHO3D_API AnimationParameters
 
     /// Empty AnimationParameters.
     static const AnimationParameters EMPTY;
+private:
```
Comment (eugeneko) [inline Source/Urho3D/Graphics/AnimationController.h:126]: Empty line above

File: Source/Urho3D/Graphics/AnimationController.h [hunk 2]
```diff
@@ -63,10 +64,22 @@ struct URHO3D_API AnimationParameters
...
+    /// @{
+    Animation* GetAnimation() const { return animation_; }
+    StringHash GetAnimationName() const { return animationName_; }
+    const WrappedScalar<float>& GetAnimationTime() const { return time_; }
+    /// @}
+
+
+    /// Time operations.
+    /// @{
+    float GetTime() const { return time_.Value(); }
```
Comment (eugeneko) [inline Source/Urho3D/Graphics/AnimationController.h:78]: We expose min time and max time in Editor already. I guess we want to have API for this as well?

File: Source/Urho3D/Graphics/AnimationController.cpp [hunk 3]
```diff
@@ -189,6 +189,16 @@ AnimationParameters& AnimationParameters::KeepOnZeroWeight()
     return *this;
 }
 
+void AnimationParameters::SetTime(float time)
```
Comment (eugeneko) [inline Source/Urho3D/Graphics/AnimationController.cpp:192]: BTW, could be inline function, especially if you add SetMinTime and SetMaxTime

File: Source/Urho3D/Graphics/AnimationController.h (hunk 2 above)
Comment (gleblebedev) [reply Source/Urho3D/Graphics/AnimationController.h:78]: You can read the time_ value via GetAnimationTime. I don't see any place where this gets written.

**Review by eugeneko — APPROVED** [review]

**Pull request merged**